src/main/java/com/sphereon/factom/identity/did/IdentityFactory.java

```diff
@@ -113,4 +167,41 @@ public DIDDocument toDid(String identifier, IdentityResponse identityResponse) t
 
         return didDocument;
     }
+
+    public DIDDocument toDid(String identifier, DidResponse didResponse) throws URISyntaxException {
+        String did = identifier;
+        if (!identifier.startsWith("did:")) {
+            did = "did:factom:" + identifier;
+        }
+        DIDURL didurl = DIDVersion.FACTOM_V1_JSON.getDidUrl(did);
+        List<DidKey> didKeys = Optional.ofNullable(didResponse.getFactomDidContent().getDidKey()).orElse(new ArrayList<>());
+        List<String> authentications = new ArrayList<>();
+        List<String> assertionMethods = new ArrayList<>();
+        List<PublicKey> publicKeys = new ArrayList<>();
+        for (DidKey key : didKeys) {
+            URI keyId = new URI(key.getId());
+            if (key.getPurpose().contains(KeyPurpose.AUTHENTICATION)) {
+                authentications.add(keyId.toString());
+            } else {
+                // ToDo: figure out what to do with assertionMethods (probably need a new model constant for KeyPurpose)
+                assertionMethods.add(keyId.toString());
+            }
+            PublicKey publicKey = PublicKey.builder()
+                    .id(keyId)
+                    .type(key.getType().getValue())
+                    .publicKeyBase58(key.getPublicKeyBase58())
+                    .build();
+            publicKey.setJsonObjectKeyValue(DIDConstants.JSONLD_TERM_CONTROLLER, key.getController());
+            publicKeys.add(publicKey);
+        }
+        DIDDocument didDocument = DIDDocument.builder()
+                .context(new URI("https://www.w3.org/ns/did/v1"))
```

Comment:
Constant

Reply:
updated, and also for the identity chain dids as well.